src/main/scala/is/hail/utils/IntervalTree.scala

```diff
@@ -147,14 +160,14 @@ object IntervalTree {
         rt.map(x => pord.min(x.minimum, min1)).getOrElse(min1)
       },
         {
-        val max1 = lft.map(x => pord.max(x.maximum, i.end)).getOrElse(i.end)
-        rt.map(x => pord.max(x.maximum, max1)).getOrElse(max1)
-      }, v))
+          val max1 = lft.map(x => pord.max(x.maximum, i.end)).getOrElse(i.end)
+          rt.map(x => pord.max(x.maximum, max1)).getOrElse(max1)
+        }, v))
     }
   }
 
   def gen[T](pord: ExtendedOrdering, pgen: Gen[T]): Gen[IntervalTree[Unit]] = {
-    Gen.buildableOf[Array](Interval.gen(pord, pgen)).map(a => IntervalTree.apply(pord, a))
+    Gen.buildableOf[Array](Interval.gen(pord, pgen)).map(a => IntervalTree.apply(pord, a.asInstanceOf[Array[Interval]]))
```

Comment:
Neither is idea but you can also write `Gen.buildableOf[Array].apply[Interval](...).map...`

Reply:
I think I can just get rid of this. I must've added it when I had all the region value stuff floating around?